hera_cal/lstbin.py

```diff
@@ -0,0 +1,943 @@
+    antenna_names = map(lambda a: "HH{}".format(a), antenna_numbers)
+
+    # get antpos and uvw
+    antenna_positions = np.array(map(lambda k: antpos[k], antenna_numbers))
+    uvw_array = np.array([antpos[k[0]] - antpos[k[1]] for k in zip(ant_1_array, ant_2_array)])
+
+    # get zenith location
+    zenith_dec_degrees = np.ones_like(baseline_array) * dec
+    zenith_ra_degrees = utils.JD2RA(time_array, longitude)
+    zenith_dec = zenith_dec_degrees * np.pi / 180
+    zenith_ra = zenith_ra_degrees * np.pi / 180
+
+    # instantiate object
+    uvd = UVData()
+
+    # assign parameters
+    params = ['Nants_data', 'Nants_telescope', 'Nbls', 'Nblts', 'Nfreqs', 'Npols', 'Nspws', 'Ntimes',
+              'ant_1_array', 'ant_2_array', 'antenna_names', 'antenna_numbers', 'baseline_array',
+              'channel_width', 'data_array', 'flag_array', 'freq_array', 'history', 'instrument',
+              'integration_time', 'lst_array', 'nsample_array', 'object_name', 'phase_type',
+              'polarization_array', 'spw_array', 'telescope_location', 'telescope_name', 'time_array',
+              'uvw_array', 'vis_units', 'antenna_positions', 'zenith_dec', 'zenith_ra']              
+    for p in params:
+        uvd.__setattr__(p, locals()[p])
+
+    # write uvdata
+    if write_miriad:
+        # check output
+        fname = os.path.join(outdir, fname)
+        if os.path.exists(fname) and overwrite is False:
+            abscal.echo("{} exists, not overwriting".format(fname), verbose=verbose)
+        else:
+            abscal.echo("saving {}".format(fname), type=0, verbose=verbose)
+            uvd.write_miriad(fname, clobber=True)
+
+    if return_uvdata:
+        return uvd
+
+
+def sigma_clip(array, flags=None, sigma=4.0, axis=0, min_N=4):
+    """
+    one-iteration sigma clipping algorithm. set clipped values to nan.
+
+    Parameters:
+    -----------
+    array : ndarray of complex visibility data. If 2D, [0] axis is samples and [1] axis is freq.
+
+    flags : ndarray matching array shape containing boolean flags. True if flagged.
+
+    sigma : float, sigma threshold to cut above
+
+    axis : int, axis of array to sigma clip
+
+    min_N : int, minimum length of array to sigma clip, below which no sigma
+                clipping is performed.
+
+    Output:
+    -------
+    array : same as input array, but clipped values have been set to np.nan
+    """
+    # ensure array is an array
+    if type(array) is not np.ndarray:
+        array = np.array(array)
+
+    # ensure array passes min_N criteria:
+    if array.shape[axis] < min_N:
+        return array
+
+    # apply flags
+    if flags is not None:
+        array[flags] *= np.nan
+
+    # get robust location
+    mean = np.nanmedian(array, axis=axis)
+
+    # get MAD
+    std = np.nanmedian(np.abs(array - mean), axis=axis) * 1.4
+
+    # set cut data to nans
+    cut = np.where(np.abs(array-mean)/std > sigma)
+    array[cut] *= np.nan
+
+    return array
+
+
+def switch_bl(key):
```

Comment:
Also, you should switch the order of the polarization (and add that to the unit test)